tests/Unit/DelegatedRoleTest.php

```diff
@@ -28,80 +28,6 @@ public function testCreateFromMetadata(): void
         self::assertFalse($this->role->terminating);
     }
 
-    /**
-     * @covers ::matchesPath
-     *
-     * @param string $target
-     *   The target name.
-     * @param array|null $paths
-     *   The path patterns to match, or null to use path hash prefixes.
-     * @param array|null $pathHashPrefixes
-     *   The path hash prefixes to match, or null to use regular paths.
-     * @param bool $willMatch
-     *   Whether the given target should match either the paths or the path hash
-     *   prefixes.
-     *
-     * @dataProvider providerMatchesRolePath
-     */
-    public function testMatchesRolePath(string $target, ?array $paths, ?array $pathHashPrefixes, bool $willMatch): void
```

Comment:
This feels like test coverage that is too valuable to lose. An easy way to prevent that is to make the comparison function a `private static` method on `TargetsMetadata`, and call it with reflection (which is hacky, but acceptable for now).

Reply:
Yeah, getting fixtures set up for this would be a bit of a bigger lift, which is why I'm proposing the lazy man's way of just putting the `roleMatchesTarget` check in a private static method that we can call here.